lib/std/Build/Step/Run.zig

```diff
@@ -240,6 +242,40 @@ pub fn addPrefixedDirectoryArg(self: *Run, prefix: []const u8, directory_source:
     directory_source.addStepDependencies(&self.step);
 }
 
+/// Add a standard `--dep-file=<deps.d>` argument for the child process to fill
+/// with its discovered additional dependencies.
+pub fn addDepFileOutputArg(self: *Run) std.Build.LazyPath {
+    return self.addPrefixedDepFileOutputArg("--dep-file=");
+}
+
+/// Add a dependencies file argument for the child process to fill with its
+/// discovered additional dependencies. Same as addDepFileOutputArg but you can
+/// specify the `flag` to use.
+/// `flag` supports both `"--some-flag="` (single argument) and `"--some-flag"`
+/// (two arguments) forms.
+pub fn addPrefixedDepFileOutputArg(self: *Run, flag: []const u8) std.Build.LazyPath {
+    const b = self.step.owner;
+
+    // Handle both `--dep-file=<deps.d>` and `--dep-file <deps.d>`
+    const two_args = !std.mem.endsWith(u8, flag, &.{'='});
+    const duped_flag = b.dupe(flag);
+
+    const dep_file = b.allocator.create(Output) catch @panic("OOM");
+    dep_file.* = .{
+        .prefix = if (two_args) "" else duped_flag,
+        .basename = "deps.d",
+        .generated_file = .{ .step = &self.step },
+    };
+
+    self.dep_output_file = dep_file;
```

Comment:
Please either assert that this value is null here, and document this limitation in the doc comments of the function, or make it handle N dep files.